server/src/main/java/org/opensearch/plugins/SecureTransportSettingsProvider.java

```diff
@@ -36,6 +37,15 @@ default Collection<TransportAdapterProvider<Transport>> getTransportAdapterProvi
         return Collections.emptyList();
     }
 
+    /**
+     * Returns true if dual mode is enabled. Dual mode domains support both encrypted and non-encrypted traffic
+     * @param settings settings
+     * @return a boolean indicating if dual mode is enabled
+     */
+    default boolean isDualModeEnabled(Settings settings) {
```

Comment:
Let's think may be in terms of secure transport parameters / settings, fe:

```suggestion
    default Optional<SecureTransportParameters> parameters(Settings settings) {
     }
     
     class SecureTransportParameters {
         boolean dualModeEnabled();
     }
```

so we could evolve it further,  wdyt?

Reply:
Agreed. I took this change, updated the usages in SecureNetty4Transport and updated the docstring